docs/tutorial.md

```diff
@@ -1,5 +1,909 @@
+view: () => Container([
+  Filter({
+    filter: "ocean"
+  }),
+  StoryList({
+    stories: {
+      "112": {
+        title: "The Ocean is Sinking",
+        author: "Kat Stropher",
+        seen: false,
+      },
+      "113": {
+        title: "Ocean life is brutal",
+        author: "Surphy McBrah",
+        seen: true,
+      },
+      "114": {
+        title: "Family friendly fun at the ocean exhibit",
+        author: "Guy Prosales",
+        seen: true,
+      }
+    },
+    reading: "113",
+    filter: "ocean"
+  }),
+  StoryDetail({
+    title: "Ocean life is brutal",
+    author: "Surphy McBrah",
+  }),
+  AutoUpdate(),
+])
+```
+
+What you see on the page should be exactly the same as before, because we haven't
+changed what `view` returns. Using basic functional composition, we were able to make
+the code a bit more manageable, and that's the only difference.
+
+State
+-------------------------------
+
+With all that view logic broken out in separate functions, `view` is starting to look like
+plain _data_. The next step is to fully separate data from the view.
+
+Add an `init` property to your app, with this pure data:
+
+```js
+  init: {
+    filter: "ocean",
+    reading: "113",
+    stories: {
+      "112": {
+        title: "The Ocean is Sinking",
+        author: "Kat Stropher",
+        seen: false,
+      },
+      "113": {
+        title: "Ocean life is brutal",
+        author: "Surphy McBrah",
+        seen: true,
+      },
+      "114": {
+        title: "Family friendly fun at the ocean exhibit",
+        author: "Guy Prosales",
+        seen: true,
+      }
+    }
+  },
+```
+
+The value of `init` becomes the app's _state_. Hyperapp calls `view` with the state
+as an argument, so it can be reduced to:
+
+```js
+  view: state => Container([
+    Filter(state),
+    StoryList(state),
+    StoryDetail(state.reading && state.stories[state.reading]),
+    AutoUpdate(state),
+  ]),
+```
+
+Visually, everything is _still_ the same.
+
+Actions
+---------------------
+
+Now that we know all about rendering views, it's finally time for some _action_!
+
+### Reacting to events in the DOM
+
+The first bit of dynamic behavior we will add is so that when you click
+the pencil-button, a text input with the filter word appears.
+
+Add an `onClick` property to the button in the filter view:
+
+```js
+const Filter = props => h("div", {class: "filter"}, [
+  "Filter:",
+  h("span", {class: "filter-word"}, props.filter),
+  h("button", { onClick: StartEditingFilter }, "\u270E") // <---
+])
+```
+
+This makes Hyperapp bind a click-event handler on the button element, so
+that when the button is clicked, an action named `StartEditingFilter` is
+_dispatched_. Create the action in the "ACTIONS" section:
+
+```js
+const StartEditingFilter = state => ({...state, editingFilter: true})
+```
+
+Actions are just functions describing transformations of the state.
+This one keep everything the same except that it sets `editingFilter` to `true`.
```

Comment:
*This action basically keeps the same state, except for `editingFilter` which it sets to `true`.